.github/workflows/test.yaml

```diff
@@ -0,0 +1,33 @@
+name: Test
+on: [push, pull_request]
```

Comment:
Running on every push is a lot, however since actions are free and we're also caching the CLion downloads to not smash Jetbrains' servers, I think it's OK.